llvm/lib/Target/AIE/AIEHazardRecognizer.h

```diff
@@ -107,6 +108,8 @@ class AIEHazardRecognizer : public ScheduleHazardRecognizer {
 
   ~AIEHazardRecognizer() override {}
 
+  AIEAlternateDescriptors SelectedAltDescs;
```

Comment:
Do we want this to be public? Why not just replace the old private attribute `std::map<MachineInstr *, unsigned> SelectedAltOpcodes;` with `AIEAlternateDescriptors SelectedAltDescs`?

Reply:
It is public because of the way I am using it in AIEMachineScheduler.cpp

Reply:
Why not just keep the existing `getSelectedAltOpcode()` method, and just change the implementation underneath? I wouldn't mind adding `AIEAlternateDescriptors SelectedAltDescs` as a public member to BlockState/RegionState, but I feel it's out of place within the HazardRecognizer.

Reply:
I would like to abstract getting Opcode or Descriptor for multiSlot or sigleSlot all under a same object.
In the follow-up work I fund have an object of `AIEAlternateDescriptors` the best way. (in #182 )

Reply:
Agreed, I prefer going with the second approach of making AIEAlternateDescriptors  a member of  BlockState/RegionState

Reply:
I have made `AIEAlternateDescriptors SelectedAltDescs` a private member of the HazardRecognizer class, and provided a member function to get a const pointer to same.

We cannot remove  `AIEAlternateDescriptors SelectedAltDescs` from hazardRecognizer without affecting the functionality of multi-slot instruction.

Reply:
Discussed offline: we'll make `AIEAlternateDescriptors` part of the `BlockState` and pass a reference to it when constructing the `HazardRecognizer`